File: core/src/main/java/kandroid/utils/CatException.java
package kandroid.utils;

public class CatException extends RuntimeException {
	public CatException() {
		super();
	}

    public CatException(String detailMessage) {
		super(detailMessage);
	}

    public CatException(String detailMessage, Throwable throwable) {
		super(detailMessage, throwable);
	}

    public CatException(Throwable throwable) {
		super(throwable);
	}
}

File: core/src/main/java/kandroid/utils/CatException.kt
package kandroid.utils

class CatException : RuntimeException {
    constructor() : super() {
    }

    constructor(detailMessage: String) : super(detailMessage) {
    }

    constructor(detailMessage: String, throwable: Throwable) : super(detailMessage, throwable) {
    }

    constructor(throwable: Throwable) : super(throwable) {
    }
}